test/unit/CodeGenerator/AttributeGeneratorTest.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\Event\CodeGenerator;
+
+use OnMoon\OpenApiServerBundle\CodeGenerator\AttributeGenerator;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\DtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\GraphDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\OperationDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\PropertyDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\RequestDtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\ResponseDtoDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\SpecificationDefinition;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\CodeGenerator\AttributeGenerator
+ */
+class AttributeGeneratorTest extends TestCase
+{
+    public function testSetAllAttributes(): void
+    {
+        $attributesGenerator = TestCase::createPartialMock(
```

Comment:
The tested class should not be mocked, because you never know what other methods can be called internally on a real class, and the test will keep passing even if those methods are broken on the real class, because everything is mocked out,

Try rewriting the test using a real AttributeGenerator with a real GraphDefinition.